### PR TITLE
[FIX] partner_second_lastname: False value should be set on lastname2

### PR DESCRIPTION
**Pull request by hugho-ad**
not firstname

before this commit the following error happen:
name = 'Van-Eyck Jan'

this was converted to

firstname = False
lastname= 'Van-Eyck'
lastname2 = 'Jan'

and it should be like

firstname = 'Jan'
lastname= 'Van-Eyck'
lastname2 = False

Before:
image

After
image

### CODE REVIEW

**Comment by luiseevaquer** [comment]
LGTM :+1:

**Comment by luiseevaquer** [comment]
@moylop260 

Could you review this PR, please?

**Comment by moylop260** [comment]
I'm not able to check it

I prefer to request review of authors:
@Yajo @pedrobaeza 
Could you check it, please?

In fact,
@hugho-ad @lescobarvx 
The CI is red

**Comment by hugho-ad** [comment]
@moylop260 
done, CI are green

**Comment by moylop260** [comment]
@Yajo @pedrobaeza 
What do you think?

**Review by pedrobaeza — APPROVED** [review]
We are not using anymore this version nor this module, but seems OK

**Comment by rousseldenis** [comment]
I would have added a test that highlight the problem.
@hugho-ad

File: partner_second_lastname/models/res_partner.py [hunk 1]
```diff
@@ -85,8 +85,8 @@ def _get_inverse_name(self, name, is_company=False):
                     parts = result['firstname'].split(" ", 1)
                 while len(parts) < 2:
                     parts.append(False)
-                result['lastname2'] = parts[0]
-                result['firstname'] = parts[1]
+                result['firstname'] = parts[0]
```
Comment (yajo) [inline partner_second_lastname/models/res_partner.py]: I think this is wrong.

Check line 83: `if order == 'last_first':`. So, I understand lastname goes first with this setting.

**Review by marcelsavegnago — APPROVED** [review]

**Comment by moylop260** [comment]
@hugho-ad @WRZ96 @lescobarvx 
Could you check the Jairo's comment, please?

File: partner_second_lastname/models/res_partner.py (hunk 1 above)
Comment (hugho-ad) [reply partner_second_lastname/models/res_partner.py]: @Yajo 

could you review again please

**Comment by WR-96** [comment]
@Yajo The method to split the last name have been refactored, could you review this PR again please

Cc: @marcelsavegnago @rousseldenis @pedrobaeza

**Comment by WR-96** [comment]
@Yajo @marcelsavegnago @rousseldenis @pedrobaeza 
Is this PR still approved after the last changes? Could you review it please

**Review by moylop260 — APPROVED** [review]
:+1:

**Comment by pedrobaeza** [comment]
@Yajo's concern is not satisfied AFAIK

**Comment by pedrobaeza** [comment]
@Yajo do you see a blocking point for reverting this?

**Comment by WR-96** [comment]
Hi @pedrobaeza, 

@Yajo's concern was about to the assign for the parts of the name according to the `partner_names_order` parameter, he said:

> I think this is wrong.
> Check line 83: if order == 'last_first':. So, I understand lastname goes first with this setting.

This is covered in the [line 93](https://github.com/OCA/partner-contact/pull/953/files#diff-e19e8b2b1ac28fd235743f835a140361R93) of `res_partner.py`, as the last name goes first we need to split the first name which contains the remaining part of the name

i.e. for a name written as 'last_name last_name2 first_name' like 'Wolowitz Joel Howard' when calling `_get_inverse_name` Odoo will return `{'lastname': 'Wolowitz', 'firstname': 'Joel Howard'}` by splitting the first name again we update the dict returned by `_get_inverse_name` which will now contain `{'lastname': 'Wolowitz', 'lastname2': 'Joel',  'firstname': 'Howard'}`

**Comment by pedrobaeza** [comment]
OK, I let him to say if it's correct, as I don't know this.

**Pull request merged**